src/main/java/com/example/webapp/Service/DepartmentService.java

```diff
@@ -0,0 +1,71 @@
+package com.example.webapp.Service;
+
+import com.example.webapp.Model.Department;
+import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.dto.DepartmentDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DepartmentService {
+    @Autowired
+    DepartmentRepository departmentRepository;
+
+    public List<DepartmentDTO> getAll() {
+        List<DepartmentDTO> departmentList = new ArrayList<>();
+        departmentRepository.findAll().forEach(department -> departmentList.add(convertENTtoDTO(department)));
+        if (departmentList.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return departmentList;
+    }
+
+    public DepartmentDTO getById(long id) {
+        Optional<Department> dep = departmentRepository.findById(id);
```

Comment:
can you add a custom exception throw if for example is not present

Reply:
Done